crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -775,6 +773,19 @@ impl CheckpointStore {
         committee: &Committee,
         effects_store: impl CausalOrder + PendCertificateForExecution,
         metrics: &CheckpointMetrics,
+    ) -> SuiResult {
+        self.check_checkpoint_transactions(contents.transactions.iter(), &effects_store)?;
+        self.process_synced_checkpoint_certificate(checkpoint, contents, committee, metrics)
+    }
+
+    /// Unlike process_new_checkpoint_certificate this does not verify that transactions are executed
+    /// Checkpoint sync process executes it because it verifies transactions when downloading checkpoint
+    pub fn process_synced_checkpoint_certificate(
```

Comment:
Why not also sync the transactions and call `process_new_checkpoint_certificate` in the other callpath as well? Then we don't need two versions

Reply:
It turns out the other path actually also sync transactions, so it should be ok to remove the check there as well. I also removed chunk from unit test that expect this check from `process_new_checkpoint_certificate` since it seems this check is not needed

Reply:
Where does the other path do the sync？

Reply:
Other path(`Action::NewCert` in `update_latest_checkpoint` to be specific) calls `get_one_checkpoint_with_contents` right before `process_new_checkpoint_certificate`

Reply:
But that just downloads the checkpoint, does it also sync the transactions?

Reply:
You are right, this is actually what `sync_checkpoint` does in the `sync_to_checkpoint`.

So in that case the path of least change would be to just keep previous version where we skip the check in case of `sync_to_checkpoint` but keep the check in the `update_latest_checkpoint`. This would unblock full node checkpoint sync but keep other code path intact.

More radical change I could add call to `sync_checkpoint` in `update_latest_checkpoint`, but I am not sure if this can cause any additional issues.

Reply:
Overall I am leaning towards keeping this PR minimal and keeping original version where we just have two functions for now, and perhaps some time later we can also have a different PR that does sync in `update_latest_checkpoint`(I can file an issue so that we don't forget). Does this make sense @lxfind ?

Reply:
Sounds good to me.
We probably should still address the error message thing though

Reply:
Done, removed error message and reverted to simplified PR version